GoogleForCreators/web-stories-wp
# Refactor safezone positioning (fix selection outside of canvas)

## Summary

Minor refactor to the safezone/dangerzone positioning to fix oddities in measurements in react-moveable and lasso selection. Most notably fixes an issue that used to shift frame boxes when selecting an element that partially sits outside of the canvas.

## Relevant Technical Choices

Moved to using `position: absolute` for a few layers to minimize side effects of adding/removing elements to the layers and used explicit pixel heights of the danger zone instead of relying on `flex-box` centering.

## Testing Instructions

1. Add an element to the canvas
2. Drag the element to the edge of the canvas such that part of the element is outside of the page
3. Unselect and reselect the element, this should show the correct box around the element, which used to be wrong.

Fixes #1787 
Fixes #1545
